### ISSUE TITLE
use monolithic marathon api

### ISSUE THREAD

**Issue by siggy** [BuoyantIO/linkerd#790]
currently the marathon namer uses a pair of marathon APIs get get service discovery information:

1. `/marathon/v2/apps`
2. `/marathon/v2/apps/[app]`

instead, just call a single api with extra params:
`/marathon/v2/apps?embed=apps.tasks`

full details:
http://mesosphere.github.io/marathon/docs/rest-api.html#get-v2-apps

proposed scala change:
`marathon/src/main/scala/io/buoyant/marathon/v2/Api.scala`:

```scala
trait Api {
  def getApps(): Future[Api.Apps]
}

object Api {
  type Apps = Map[Path, Addr]
}
```

taking this a step further, consider using server side events (SSE), or marathon api callbacks.

relates to #105 and #704

**Comment by olix0r** [comment]
per discussion: we're a little concerned that a single pull of the entire dataset could be much heavier resource-wise.  We'd want to be ensure that this system works well in non-trivial deployments.

Ideally, of course, marathon would provide a non-SSE streaming or watch API.

**Comment by moderation** [comment]
https://mesosphere.github.io/marathon/docs/generated/api.html#v2_eventSubscriptions_get ?

**Comment by olix0r** [comment]
@moderation yeah, this is another approach that could work well especially for namerd.  I'm reluctant to rely on callbacks to hit all linkerd instances in a cluster...
